tutorials/plot_metadata_epochs.py

```diff
@@ -69,36 +71,76 @@
-    query = 'NumberOfLetters == {} and is_concrete == "{}"'.format(cat1, cat2)
-    this_epochs = epochs[query]
-    avs["{}/{}".format(cat1, cat2)] = this_epochs.average()
-
-# Style the plot
-colors = np.linspace(0, 1, num=len(avs))
+# Create two new metadata columns
+metadata = epochs.metadata
+is_concrete = metadata["Concreteness"] > metadata["Concreteness"].median()
+metadata["is_concrete"] = np.where(is_concrete, 'Concrete', 'Abstract')
+is_concrete = metadata["NumberOfLetters"] > 5
+metadata["is_long"] = np.where(is_concrete, 'Long', 'Short')
+epochs.metadata = metadata
+
+###############################################################################
+# Now we can quickly extract (and plot) subsets of the data. For example, to
+# look at words split by word length and concreteness:
+
+query = "is_long == '{0}' & is_concrete == '{1}'"
+evokeds = dict()
+for concreteness in ("Concrete", "Abstract"):
+    for length in ("Long", "Short"):
+        subset = epochs[query.format(length, concreteness)]
+        evokeds["/".join((concreteness, length))] = list(subset.iter_evoked())
+
+# For the actual visualisation, we store a number of shared parameters.
 style_plot = dict(
-    colors=plt.get_cmap('YlGnBu_r')(colors),
-    linestyles={'Concrete': '-', 'Abstract': '--'}
+    colors={"Long": "Crimson", "Short": "Cornflowerblue"},
+    linestyles={"Concrete": "-", "Abstract": ":"},
+    split_legend=True,
+    ci=.68,
+    show_sensors=4,
+    show_legend=3,
+    truncate_yaxis="max_ticks",
+    picks=epochs.ch_names.index("Pz"),
 )
 
-# Make the plot
-ix_plot = mne.pick_channels(epochs.ch_names, ['Pz'])
-fig, ax = plt.subplots(figsize=(6, 3))
-fig = mne.viz.evoked.plot_compare_evokeds(
-    avs, picks=ix_plot, show=False, axes=ax, **style_plot)
-ax.legend(loc=[1.05, .1])
+fig, ax = plt.subplots(figsize=(6, 4))
+mne.viz.plot_compare_evokeds(evokeds, axes=ax, **style_plot)
 plt.show()
 
 ###############################################################################
+# To compare words which are 4, 5, 6, 7 or 8 letters long:
+
+evokeds = dict()
+for n_letters in epochs.metadata["NumberOfLetters"].unique():
+    n_letters = str(n_letters)
+    evokeds[n_letters] = epochs["NumberOfLetters == " + n_letters].average()
+
+style_plot["colors"] = {str(n_letters): int(n_letters) for n_letters in
+                        epochs.metadata["NumberOfLetters"].unique()}
+style_plot["cmap"] = ("Number of letters", "summer_r")
+del style_plot['linestyles']
+
+fig, ax = plt.subplots(figsize=(6, 4))
+mne.viz.plot_compare_evokeds(evokeds, axes=ax, **style_plot)
+plt.show()
+
+###############################################################################
+# And finally, for the interaction between concreteness and continuous length
+# in letters:
+evokeds = dict()
+query = "is_concrete == '{0}' & NumberOfLetters == {1}"
+for concreteness in ("Concrete", "Abstract"):
+    for n_letters in epochs.metadata["NumberOfLetters"].unique():
```

Comment:
`epochs.metadata["NumberOfLetters"].unique()` appears a number of times in the example, might be worth creating a variable for that - `num_letters_unique` or somthing like that.